### PR TITLE
clarify security fixes policy

### LINKED ISSUE #9323: Clarify whether assistance support includes security patches
On our [release support policy page](https://www.cockroachlabs.com/docs/releases/release-support-policy.html), we outline the different levels of support we offer for CockroachDB releases. It is not currently clear from this page whether we will continue to offer security patches for releases in the "assistance support" window. We should add a bullet that indicates that we _do_ still offer critical security fixes to all releases in the support window, including releases in the "assistance support" window.

### PR DESCRIPTION
**Pull request by taroface**
Fixes #9323.

### CODE REVIEW

**Comment by taroface** [comment]
@piyush-singh Friendly ping on review

**Review by piyush-singh — APPROVED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9335)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @taroface)

File: releases/release-support-policy.md [hunk 1]
```diff
@@ -10,12 +10,13 @@ This page explains Cockroach Lab's policy for supporting [major releases](../rel
 
 Each major release of CockroachDB goes through the following support cycle:
 
-- **Maintenance Support:** For at least 365 days from the major release date, Cockroach Labs will produce regular patch releases including resolutions to problems identified by users.
+- **Maintenance Support:** For at least 365 days from the major release date, Cockroach Labs will produce regular patch releases including resolutions to problems identified by users and critical security fixes.
```
Comment (mikeCRL) [inline releases/release-support-policy.md:13]:
```suggestion
- **Maintenance Support:** For at least 365 days from the major release date, Cockroach Labs will produce regular patch releases that include critical security fixes and resolutions to problems identified by users.
```

Minor edit to resolve some slight ambiguity, for your consideration.

File: releases/release-support-policy.md [hunk 2]
```diff
@@ -10,12 +10,13 @@ This page explains Cockroach Lab's policy for supporting [major releases](../rel
...
 
 - **Assistance Support:** Following the maintenance support period, Cockroach Labs will provide assistance support for at least an additional 180 days. During this period, the following guidelines will apply:
     - New enhancements and error corrections will not be made to the major release.
     - Cockroach Labs will direct customers to existing fixes/patches and workarounds applicable to the reported case.
     - Cockroach Labs may direct customers to [upgrade](../stable/upgrade-cockroach-version.html) to a more current version of the product if a workaround does not exist.
+    - Cockroach Labs will continue to add critical security fixes to the major release.
```
Comment (mikeCRL) [inline releases/release-support-policy.md:19]:
```suggestion
    - Cockroach Labs will continue to add critical security fixes to the major release in the form of patch releases.
```

A clarification, assuming my interpretation is correct. (E.g. we wouldn't actually add the fix to v20.2.0.) I reviewed the public [CockroachDB Release Cycle](https://wiki.crdb.io/wiki/spaces/CRDB/pages/68616452/CockroachDB+Release+Cycle) page to try to confirm.

**Review by mikeCRL — COMMENTED** [review]
LGTM. Suggested a couple minor clarifications.

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/9335)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mikeCRL)

___
*[releases/release-support-policy.md, line 13 at r1](https://reviewable.io/reviews/cockroachdb/docs/9335#-MR1SkJktn6Bmm5zcHMK-r1-13:-MT87cDxA7nXMQzFs0Dj:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76a03752cf272d7e7fee4c0bbc2c8071ba68ef20/releases/release-support-policy.md#L13)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
- **Maintenance Support:** For at least 365 days from the major release date, Cockroach Labs will produce regular patch releases that include critical security fixes and resolutions to problems identified by users.
```

Minor edit to resolve some slight ambiguity, for your consideration.
</blockquote></details>

Done.
___
*[releases/release-support-policy.md, line 19 at r1](https://reviewable.io/reviews/cockroachdb/docs/9335#-MR1SkJktn6Bmm5zcHMK-r1-19:-MT87qpU6ub-Hc5yisTJ:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/76a03752cf272d7e7fee4c0bbc2c8071ba68ef20/releases/release-support-policy.md#L19)):*
<details><summary><i>Previously, mikeCRL (Mike Lewis) wrote…</i></summary><blockquote>

```suggestion
    - Cockroach Labs will continue to add critical security fixes to the major release in the form of patch releases.
```

A clarification, assuming my interpretation is correct. (E.g. we wouldn't actually add the fix to v20.2.0.) I reviewed the public [CockroachDB Release Cycle](https://wiki.crdb.io/wiki/spaces/CRDB/pages/68616452/CockroachDB+Release+Cycle) page to try to confirm.
</blockquote></details>

Done.

**Comment by taroface** [comment]
TFTR, very helpful tweaks.

**Pull request merged**